fixups/EnvVarFixup/EnvVarFixup.cpp

```diff
@@ -54,156 +59,229 @@ DWORD __stdcall GetEnvironmentVariableFixup(_In_ const CharC* lpName, _Inout_ Ch
                 if (std::regex_match(eName, spec.variablename))
                 {
                     size_t valuelen = spec.variablevalue.length();
-                    if (spec.useregistry == true)
+                    if (spec.remediationType == Env_Remediation_Type_Registry)
                     {
-                        Log("[%d] GetEnvironmentVariableFixup: Registry supplied case.", GetEnvVarInstance);
-                        // Check app registry for an answer instead of the Json. Note: this allows value to be modified possibly also.
-                        HKEY hKeyCU;
-                        if constexpr (psf::is_ansi<CharT>)
+                        if (spec.useregistry == true)
                         {
-                            if (RegOpenKeyExA(HKEY_CURRENT_USER, "Environment",
-                                0, KEY_ENUMERATE_SUB_KEYS | KEY_READ | KEY_QUERY_VALUE, &hKeyCU) == ERROR_SUCCESS)
+                            Log("[%d] GetEnvironmentVariableFixup: Registry supplied case.", GetEnvVarInstance);
+                            // Check app registry for an answer instead of the Json. Note: this allows value to be modified possibly also.
+                            HKEY hKeyCU;
+                            if constexpr (psf::is_ansi<CharT>)
                             {
-                                Log("[%d] GetEnvironmentVariableFixup:(A) HKCU key found.", GetEnvVarInstance);
-                                DWORD type = RRF_RT_REG_SZ;
-                                DWORD dLen = (DWORD)lenBuf;
-                                auto ret = RegGetValueA(HKEY_CURRENT_USER, "Environment", lpName,
-                                    RRF_RT_REG_SZ | RRF_RT_REG_EXPAND_SZ | RRF_ZEROONFAILURE, &type, lpValue, &dLen);
-                                if (ret == ERROR_SUCCESS)
+                                if (RegOpenKeyExA(HKEY_CURRENT_USER, "Environment",
+                                    0, KEY_ENUMERATE_SUB_KEYS | KEY_READ | KEY_QUERY_VALUE, &hKeyCU) == ERROR_SUCCESS)
                                 {
-                                    Log("[%d] GetEnvironmentVariableFixup:(A) HKCU value found %s", GetEnvVarInstance, lpValue);
-                                    result = dLen;
+                                    Log("[%d] GetEnvironmentVariableFixup:(A) HKCU key found.", GetEnvVarInstance);
+                                    DWORD type = RRF_RT_REG_SZ;
+                                    DWORD dLen = (DWORD)lenBuf;
+                                    auto ret = RegGetValueA(HKEY_CURRENT_USER, "Environment", lpName,
+                                        RRF_RT_REG_SZ | RRF_RT_REG_EXPAND_SZ | RRF_ZEROONFAILURE, &type, lpValue, &dLen);
+                                    if (ret == ERROR_SUCCESS)

```

Comment:
Please use WIL macro for testing error codes.

Reply:
Registry apis return LSTATUS not HRESULT